### PR TITLE
Rover: tidy GCS_MAVLink handleMessage function

### PR DESCRIPTION
**Pull request by peterbarker**
Co-authored-by: nubcaker

This replaces https://github.com/ArduPilot/ardupilot/pull/13821

It does not reindent the code, which makes scanning through the code to ensure no unintended logic changes have been made easy to do.

A follow-up PR could reindent the code ('though this has the bad side-effect of clouding the line-history in the file).

### CODE REVIEW

**Comment by peterbarker** [comment]
I've tested this with a combination of new autotests and running the code in SITL, adding breakpoints to ensure we still run through the packet handling code when we receive those packet.

File: Tools/autotest/rover.py [hunk 1]
```diff
@@ -5347,6 +5349,84 @@ def test_slew_rate(self):
...
+                0,     # acc-z
+                0,     # yaw
+                0,     # yaw rate
+            )
+
+            msg = self.mav.recv_match(type='VFR_HUD', blocking=True, timeout=1)
+            if msg is None:
+                raise NotAchievedException("No VFR_HUD message")
+            self.progress("speed=%f" % msg.groundspeed)
+            if msg.groundspeed > 5:
```
Comment (khancyr) [inline Tools/autotest/rover.py:5426]: Hum this is a wrong test... You should do like I have done on global_ned and check that the right position is reached.
What we want to check here is the position controler not the speed one

**Review by khancyr — COMMENTED** [review]
Seems good. 
We could probably rewamp the test to be complet later

File: Tools/autotest/rover.py (hunk 1 above)
Comment (peterbarker) [reply Tools/autotest/rover.py:5426]: Yeah, in this case I only wanted to make sure we *acted* on the message, given the nature of the PR.

I *believe* you have a pending PR which adds all sorts of cool tests for this and didn't want to replicate it!

**Comment by rmackay9** [comment]
I'd prefer if we fix the indenting as well..

File: Tools/autotest/rover.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/rover.py:5426]: ok then ! we can do better later. I don't have start working on those test yet but that is a matter of time

**Review by khancyr — APPROVED** [review]

**Comment by peterbarker** [comment]
@rmackay9 I've pushed a patch on top which does the re-indenting.  302 lines changes.

**Comment by amilcarlucas** [comment]
Most of this can, at some point, be moved to AP_Vehicle right?

**Comment by khancyr** [comment]
No most use too specific function or variable.
It would need more unification on interfaces for put those into ap_vehicle.

**Comment by peterbarker** [comment]
> Most of this can, at some point, be moved to AP_Vehicle right?

heartbeat handling can probable move up and we can have a `gcs().last_mygcs_heartbeat()` method.

hil handling should just die in a hole.

radio handling *could* move up, but we'd still need to provide a callback onto the GCS subclass to return the logging bitmask bit which tells us whether we should log the message.

The other methods are really, really vehicle-specific, especially when you've got thrust-vs-throttle stuff going on... I'm pretty sure we're always going to handle these messages as part of the vehicle code (the presence of the guided-mode stuff on vehicle for scripting NOT being a great precedent... (`methods to control vehicle for use by scripting`))

**Pull request merged**